demos/include/transport_plaintext.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef TRANSPORT_PLAINTEXT_H
+#define TRANSPORT_PLAINTEXT_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+int Transport_ConnectToServer( const char * pServer, uint16_t port );
```

Comment:
(Based on discussion with Dan and Gaurav ) Transport layer should not be an abstraction but just an interface. I.e. only two interface implementations transport_read and transport_write must be passed to the MQTT or HTTP library. create/connect/disconnect operations will be handled by application, i.e., in our case demos and they should not be abstracted out.  Also each demo will have its own create/connect/disconnect sequence.